tests/unit/components/Editor/ItemEditor.spec.js

```diff
@@ -2,7 +2,590 @@ import { mount } from "@vue/test-utils";
+        .find('[data-test="input"]').element.value
+    ).toBe("300");
+  });
+
+  it("triggers keyup", async () => {
+    const maxCharLimitInputKeypress = jest.spyOn(
+      ItemEditor.methods,
+      "maxCharLimitInputKeypress"
+    );
+    const wrapper = mount(ItemEditor, {
+      props: {
+        itemList: [
+          {
+            type: "question",
+            details: {
+              text: "test",
+              type: "subjective",
+              has_char_limit: true,
+            },
+            time: 13,
+          },
+        ],
+        questionTypeIndex: 1,
+      },
+    });
+    await wrapper
+      .find('[data-test="maxCharLimit"]')
+      .find('[data-test="input"]')
+      .trigger("keypress", {
+        key: 1,
+      });
+    expect(maxCharLimitInputKeypress).toHaveBeenCalled();
+  });
+
+  it("triggers keydown", () => {
+    const maxCharLimitInputKeydown = jest.spyOn(
+      ItemEditor.methods,
+      "maxCharLimitInputKeydown"
+    );
+    const wrapper = mount(ItemEditor, {
+      props: {
+        itemList: [
+          {
+            type: "question",
+            details: {
+              text: "test",
+              type: "subjective",
+              has_char_limit: true,
+            },
+            time: 13,
+          },
+        ],
+        questionTypeIndex: 1,
+      },
+    });
+    wrapper
+      .find('[data-test="maxCharLimit"]')
+      .find('[data-test="input"]')
+      .trigger("keydown", {
+        key: 1,
+      });
+    expect(maxCharLimitInputKeydown).toHaveBeenCalled();
+  });
+
+  it("changes question type", async () => {
+    const wrapper = mount(ItemEditor, {
+      props: {
+        itemList: [
+          {
+            type: "question",
+            details: {
+              text: "test",
+              type: "subjective",
+              has_char_limit: true,
+            },
+            time: 13,
+          },
+        ],
+        questionTypeIndex: 1,
+      },
+    });
+    // open the dropdown
+    await wrapper
+      .find('[data-test="questionTypeDropdown"]')
+      .find('[data-test="toggleButton"]')
+      .trigger("click");
+
+    // select mcq
+    await wrapper
+      .find('[data-test="questionTypeDropdown"]')
+      .find('[data-test="option-0"]')
+      .trigger("click");
+    expect(wrapper.emitted()["question-type-changed"][0][0]).toBe("mcq");
+  });
+
+  it("updates question text", async () => {
+    const wrapper = mount(ItemEditor, {
+      props: {
+        itemList: [
+          {
+            type: "question",
+            details: {
+              text: "test",
+              type: "mcq",
+            },
+            time: 13,
+          },
+        ],
+      },
+    });
+    const testValue = "test";
+    await wrapper
+      .find('[data-test="questionText"]')
+      .find('[data-test="input"]')
+      .setValue(testValue);
+    expect(wrapper.vm.localItemList[0].details.text).toBe(testValue);
+  });
+
+  it("updates max char limit", async () => {
```

Comment:
Feels like we're "updating" the max char limit value but in reality, we're unchecking the checkbox. Maybe can rename it. Maybe `enables/disables max char limit` or something like that